docs/client-side/using-gulp.rst

```diff
@@ -0,0 +1,219 @@
+
+-	Compile LESS or SASS files to CSS.
+-	Compile CoffeeScript or TypeScript files to JS.
+-	Bundle and minify your JS files.
+-	Run tools like JSHint.
+
+A *task runner* is an app that automates these routine development tasks. Visual Studio 2015 provides built-in support for two popular JavaScript-based task runners, Gulp and Grunt. 
+
+Using Gulp
+----------
+Here are the general steps to use Gulp in your ASP.NET project. This topic will cover each in more detail.
+
+1.	Use NPM to add the Gulp.js package.
+
+#.	Use NPM to add one or more `Gulp plugins <http://gulpjs.com/plugins>`_. By itself, Gulp is just an engine for running tasks. Plugins are the modules that actually do the work.
+
+#.	Define Gulp tasks.
+
+#.	Run tasks in Task Runner Explorer.
+
+#.	Bind tasks to build events.
+
+Empty vs. Web Site Projects
+---------------------------
+New ASP.NET Web Applications created in Visual Studio 2015 start with the *Empty* or *Web Site* templates. 
+ 
+-  The *Empty* Visual Studio ASP.NET template does not include NPM or Gulp configuration. You can use the Empty template to learn how NPM and Gulp work, step-by-step. 
+
+-  The *Web Site* template includes NPM and Gulp by default. NPM is preconfigured to add the Gulp.js package. Gulp is configured to copy Bootstrap, Hammer and jQuery client libraries to your web app directory.       
+
+Add the Gulp package
+--------------------
+NPM (Node Package Manager) is a package manager that was originally created for Node.js. We'll use NPM to install the Gulp.js package.
+Add an NPM configuration file.
+
+1.	In Solution Explorer, right-click the project.
+
+#.	Select Add > New Item.
+
+#.	Select **NPM configuration file**.
+
+#.	Leave the default name, "package.json".
+
+#.	Click **Add**.
+
+Open package.json. In the dependencies section, add an entry for "gulp":
+
+.. code-block:: javascript
+   :emphasize-lines: 6
+
+	{
+	  "version": "1.0.0",
+	  "name": "MyWebApp",
+	  "private": true,
+	  "devDependencies": {
+	    "gulp": "3.9.0"
+	  }
+	}
+
+The packages are identified using the numbering scheme **<major>.<minor>.<patch>** according to the `SemVer <http://semver.org/>`_ (semantic versioning) standard.
+
+After typing the name of the package and a colon, Intellisense shows a simplified list of versions with the most common choices:  
+
+.. image:: using-gulp/_static/semver.png
+
+-  The top item in the Intellisense list is considered the latest stable version of the package. 
+-  The carat ^ symbol matches the most recent major version.
+-  The tilde ~ matches the most recent minor version. 
+
+Add the Gulp config file
+------------------------
+1.	In Solution Explorer, right-click the project.
+
+#.	Select Add > New Item.
+
+#.	Select **Gulp Configuration file**.
+
+#.	Leave the default name, "gulpfile.js".
+
+#.	Click **Add**.
+
+#.  Right-click Dependencies > NPM and select **Restore Packages** from the context menu.
+
```

Comment:
Maybe add a transition sentence to say, now you're ready to define some gulp tasks .... (because the next few sections are examples of using Gulp)